### PR TITLE
Respect user display name order

### PR DESCRIPTION
**Pull request by abcdev**
return contact full name depending on user system-wide preference.

### CODE REVIEW

File: src/Contacts/Contact.m [hunk 1]
```diff
@@ -37,12 +37,17 @@ - (instancetype)initWithContactWithFirstName:(NSString *)firstName
 
 - (NSString *)fullName {
     NSMutableString *fullName = [NSMutableString string];
-    if (self.firstName)
+    if (self.firstName) {
         [fullName appendString:self.firstName];
+        }
     if (self.lastName) {
```
Comment (michaelkirk) [inline src/Contacts/Contact.m:43]: nit: indent on line #42

**Comment by michaelkirk** [comment]
Thanks for working on this.

Since we're now trimming, I think this whole thing would be a little more clear as just:

```
 - (NSString *)fullName
{

    NSString *fullstring;
    if (ABPersonGetCompositeNameFormat() == kABPersonCompositeNameFormatFirstNameFirst) {
        fullName = [NString stringWithFormat:"%@ %@", firstName, lastName];
    } else {
        fullName = [NString stringWithFormat:"%@ %@", lastName, firstName];
    }

    // Trim whitespace from missing name components
    return [fullName stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
}
```

**Comment by abcdev** [comment]
If one of the name components is nil you will end up with "(null)" in the string. Making stringByTrimmingCharactersInSet useless. As far as I understood if there is anything contact name related but not first name it ends up being in the lastName signal contact hence the leading "if (self.firstName)"

**Comment by roostr** [comment]
This work was folded into the now merged #67, so this PR is no longer valid. Thanks for the inspiration on a clean way to do the formatting, @abcdev.

**Comment by michaelkirk** [comment]
Thanks @roostr, and sorry to let yours go stale @abcdev.

**Pull request closed without merging**